### PR TITLE
Auth, environment and resources

### PR DESCRIPTION
**Pull request by katrogan**

### CODE REVIEW

File: flytekit/annotated/container_task.py [hunk 1]
```diff
@@ -63,7 +63,7 @@ def execute(self, **kwargs) -> Any:
         return None
 
     def get_container(self, settings: RegistrationSettings) -> _task_model.Container:
-        env = settings.env
+        env = {**settings.env, **self.environment} if self.environment else settings.env
```
Comment (wild-endeavor) [inline flytekit/annotated/container_task.py:66]: this works?  cool, new to me.  and the second one takes precedence?

**Review by wild-endeavor — COMMENTED** [review]
can you add one of these tests
https://github.com/lyft/flytekit/blob/annotations/tests/flytekit/unit/annotated/test_type_hints.py#L435

like sdk_task = t.get_registerable_entity() and then confirm it's passed to the sdktask as well?

File: flytekit/annotated/container_task.py (hunk 1 above)
Comment (katrogan) [reply flytekit/annotated/container_task.py:66]: this adds them together. is that undesirable?

Comment (wild-endeavor) [reply flytekit/annotated/container_task.py:66]: that is desirable, just wondering what happens if the same key is in both.

**Comment by katrogan** [comment]
@wild-endeavor 

> can you add one of these tests

done

**Pull request merged**